src/Security/TokenChecker.php

```diff
@@ -25,55 +26,72 @@ class TokenChecker
     private $session;
 
     /**
-     * @param SessionInterface $session
+     * @var AuthenticationTrustResolverInterface
      */
-    public function __construct(SessionInterface $session)
+    private $trustResolveer;
+
+    /**
+     * @param SessionInterface                     $session
+     * @param AuthenticationTrustResolverInterface $trustResolveer
+     */
+    public function __construct(SessionInterface $session, AuthenticationTrustResolverInterface $trustResolveer)
     {
         $this->session = $session;
+        $this->trustResolveer = $trustResolveer;
     }
 
     /**
-     * Checks if an authenticated token exists in the session.
-     *
-     * @param string $sessionKey
+     * Checks if a front end user is authenticated.
      *
      * @return bool
      */
-    public function hasAuthenticatedToken(string $sessionKey): bool
+    public function hasFrontendUser(): bool
     {
-        $token = $this->getToken($sessionKey);
+        $token = $this->getToken(FrontendUser::SECURITY_SESSION_KEY);
 
-        return null !== $token && $token->getUser() instanceof User;
+        return null !== $token && $token->getUser() instanceof FrontendUser;
     }
 
     /**
-     * Gets the username of a token in the session.
+     * Checks if a back end user is authenticated.
      *
-     * @param string $sessionKey
-     *
-     * @return string|null
+     * @return bool
      */
-    public function getUsername(string $sessionKey): ?string
+    public function hasBackendUser(): bool
     {
-        $token = $this->getToken($sessionKey);
+        $token = $this->getToken(BackendUser::SECURITY_SESSION_KEY);
 
-        if (null === $token || !$token->getUser() instanceof User) {
-            return null;
-        }
+        return null !== $token && $token->getUser() instanceof BackendUser;
+    }
 
-        return $token->getUser()->getUsername();
+    /**
+     * Gets the front end username from the session.
+     *
+     * @return string|null
+     */
+    public function getFrontendUsername(): ?string
+    {
+        return $this->getUsername(FrontendUser::class);
     }
 
     /**
-     * Checks whether the front end preview is active.
+     * Gets the back end username from the session.
      *
-     * @param string $sessionKey
+     * @return string|null
+     */
+    public function getBackendUsername(): ?string
+    {
+        return $this->getUsername(BackendUser::class);
+    }
+
+    /**
+     * Tells whether the front end preview can show unpublished fragments.
      *
      * @return bool
      */
-    public function isPreviewMode(string $sessionKey): bool
+    public function showUnpublished(): bool
```

Comment:
The naming is wrong. Methods returning booleans have to start with `is`, `has`, `should` etc.

Reply:
what would you use instead? `shouldShowUnpublished`?

Reply:
`isPreviewMode()` was fine with me. Similar to `Model::isPreviewMode()`.

Reply:
Just posting here again for reference: 
> There is a distinction between preview mode (using the frontend preview with frames) and showing unpublished items. The FrontendPreviewToken is used to login a temporary user, but the unpublished elements can still be hidden.